### PR TITLE
Add twig function for sso params

### PR DESCRIPTION
**Pull request by ricsam**

### CODE REVIEW

File: Twig/Auth0Extension.php [hunk 1]
```diff
@@ -41,10 +41,31 @@ public function getFunctions()
...
             new \Twig_SimpleFunction('state_parameter', [$this, 'stateParameter']),
+            new \Twig_SimpleFunction('sso_params', [$this, 'ssoParams']),
             new \Twig_SimpleFunction('sso_login_url', [$this, 'loginUrl']),
         ];
     }
 
+    public function ssoParams()
+    {
+
+        if ($this->csrfTokenManager && $this->generator) {
```
Comment (magnusnordlander) [inline Twig/Auth0Extension.php]: `$this->generator` is required in the constructor, and thus always not null.

File: Twig/Auth0Extension.php [hunk 2]
```diff
@@ -41,10 +41,31 @@ public function getFunctions()
...
+            $csrfToken = $this->csrfTokenManager->getToken('auth0-sso');
+
+            $token = $csrfToken->getValue();
+            $domain = $this->generator->getAuth0Domain();
+            $client_id = $this->generator->getAuth0ClientId();
+            return [
+                'token' => $token,
```
Comment (magnusnordlander) [inline Twig/Auth0Extension.php]: This should probably be renamed `csrf_token`.

File: SSOUrlGenerator.php [hunk 3]
```diff
@@ -51,4 +51,12 @@ public function generateUrl($redirectUri, $options = [])
 
         return sprintf('https://%s/authorize?%s', $this->auth0Domain, http_build_query($query));
     }
+    public function getAuth0Domain()
```
Comment (magnusnordlander) [inline SSOUrlGenerator.php]: Add a blank line between methods.

**Review by magnusnordlander — CHANGES_REQUESTED** [review]

**Comment by ricsam** [comment]
@magnusnordlander Thanks 🦀! Can you merge this if you think it is okey?

File: SSOUrlGenerator.php [hunk 4]
```diff
@@ -51,4 +51,14 @@ public function generateUrl($redirectUri, $options = [])
 
         return sprintf('https://%s/authorize?%s', $this->auth0Domain, http_build_query($query));
     }
+
```
Comment (magnusnordlander) [inline SSOUrlGenerator.php]: I don't think it makes sense for the URL generator to have these getters, so remove them...

File: Twig/Auth0Extension.php [hunk 5]
```diff
@@ -41,10 +41,29 @@ public function getFunctions()
...
         ];
     }
 
+    public function ssoParams()
+    {
+        if ($this->csrfTokenManager) {
+            $csrfToken = $this->csrfTokenManager->getToken('auth0-sso');
+
+            $csrf_token = $csrfToken->getValue();
+            $domain = $this->generator->getAuth0Domain();
```
Comment (magnusnordlander) [inline Twig/Auth0Extension.php]: ... and add the parameters as constructor arguments in the extension. You'll also need to add them to the service definition in Resouces/config/services.yml.

**Review by magnusnordlander — CHANGES_REQUESTED** [review]

**Review by magnusnordlander — APPROVED** [review]

**Pull request merged**